BaseLib/FileTools.cpp

```diff
@@ -39,6 +39,107 @@ bool IsFileExisting(const std::string &strFilename)
     return (stat (strFilename.c_str(), &buffer) == 0);
 }
 
+std::tuple<std::string, std::string::size_type, std::string::size_type>
+getParenthesizedString(std::string const& in,
+                       char const open_char,
+                       char const close_char,
+                       std::string::size_type pos)
+{
+    auto const pos_curly_brace_open = in.find_first_of(open_char, pos);
+    if (pos_curly_brace_open != std::string::npos)
+    {
+        auto const pos_curly_brace_close =
+            in.find_first_of(close_char, pos_curly_brace_open);
+        if (pos_curly_brace_close != std::string::npos)
+        {
+            return std::make_tuple(
+                in.substr(pos_curly_brace_open + 1,
+                          pos_curly_brace_close - (pos_curly_brace_open + 1)),
+                pos_curly_brace_open, pos_curly_brace_close);
+        }
+    }
+    return std::make_tuple("", std::string::npos, std::string::npos);
+}
+
+std::pair<bool, std::string> containsKeyword(std::string const& str,
```

Comment:
The return type allows empty keywords to be valid return values.
Better to return a string. If string is empty => not valid.

Reply:
Fixed.